test/python/pulse/test_transforms.py

```diff
@@ -401,12 +407,286 @@ def test_multiple_schedules(self):
+        compressed_schedule = transforms.compress_pulses(schedules)
         original_pulse_ids = get_pulse_ids(schedules)
         compressed_pulse_ids = get_pulse_ids(compressed_schedule)
         self.assertEqual(len(original_pulse_ids), 6)
         self.assertEqual(len(compressed_pulse_ids), 2)
 
 
+class TestAlignSequential(QiskitTestCase):
+    """Test sequential alignment transform."""
+
+    def test_align_sequential(self):
+        """Test sequential alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_sequential_with_barrier(self):
+        """Test sequential alignment with a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule.append(directives.RelativeBarrier(d0, d1), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(
+            transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+
+class TestAlignLeft(QiskitTestCase):
+    """Test left alignment transform."""
+
+    def test_align_right(self):
+        """Test left alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
+
+        schedule = transforms.align_left(schedule)
```

Comment:
I thought `align_left` only affected the top level schedules? Not the case? If I add a cnot schedule using `align_left`, it will change the timing of the internal operations, right? (i.e. we'd have to use some other method for that)

Reply:
This could also be rewritten so that equivalent instructions appear in the same relative order, to help readers and maintainers

Reply:
> I thought `align_left` only affected the top level schedules? Not the case? If I add a cnot schedule using `align_left`, it will change the timing of the internal operations, right? (i.e. we'd have to use some other method for that)

The internal timing of operations *are* maintained.

> This could also be rewritten so that equivalent instructions appear in the same relative order, to help readers and maintainers

When writing and maintaining these tests I find it easier to consider the outcome of scheduling on a per-channel basis as the alignment operations can rearrange the ordering.